File: libs/ui/announcement-reactions.ts
export const announcementReactions = [
  { key: "love", emoji: "🥰", label: "Love" },
  { key: "wow", emoji: "🤩", label: "Wow" },
  { key: "confused", emoji: "😕", label: "Confused" },
  { key: "support", emoji: "🥺", label: "Support" },
  { key: "happy", emoji: "😄", label: "Happy" },
] as const;

export type AnnouncementReaction = typeof announcementReactions[number]["key"];
export type AnnouncementReactions = {
  counts: Record<AnnouncementReaction, number>;
  choice: AnnouncementReaction | null;
  total: number;
};

export function emptyReactions(): AnnouncementReactions {
  return { counts: { love: 0, wow: 0, confused: 0, support: 0, happy: 0 }, choice: null, total: 0 };
}

File: src/atlas/MorganHacks.Applications/Services/AnnouncementReactions.cs
namespace MorganHacks.Applications.Services;

public sealed record AnnouncementReactions(Dictionary<string, int> Counts, string? Choice)
{
    public int Total => Counts.Values.Sum();

    public static bool IsValid(string reaction) => reaction is "love" or "wow" or "confused" or "support" or "happy";

    public static AnnouncementReactions Empty() => new(new()
    {
        ["love"] = 0,
        ["wow"] = 0,
        ["confused"] = 0,
        ["support"] = 0,
        ["happy"] = 0,
    }, null);
}
